### ISSUE TITLE
F1 values of WHU-CD and DSIFN-CD

### ISSUE THREAD

**Issue by yuwanting828** [justchenhao/BIT_CD#28]
Hello, I am trying to reproduce your model code, and it looks incorrect when using WHU-CD dataset as well as DSIFN-CD dataset, and the result appears too high than your result. But it looks normal when using LEVID-CD dataset. Do I need to change the pre-training?

**Comment by YangFanghan** [comment]
Can you share with me the training hyperparameters on the dsifn dataset, my training results are very different from the original text, and there is no best model after 60 epoch, this is my email, can you share the dsifn data collection, and your training methods and results。

**Comment by yuwanting828** [comment]
Hello,can you share me with your training results?
and my  dsifn dataset is downloaded from https://github.com/wgcban/ChangeFormer.You can read a lot from that, I hope my advice is useful to you

**Comment by YangFanghan** [comment]
gpu_ids: [0, 3] project_name: CD_bit_b16_lr0.01 checkpoint_root: checkpoints-test num_workers: 4 dataset: CDDataset data_name: DSIFN-CD-ori batch_size: 16 split: train split_val: val img_size: 512 n_class: 2 net_G: base_transformer_pos_s4_dd8_dedim8 loss: ce optimizer: sgd lr: 0.01 max_epochs: 200 lr_policy: linear lr_decay_iters: 100 checkpoint_dir: checkpoints-test/CD_bit_b16_lr0.01 vis_dir: vis/CD_bit_b16_lr0.01 loading last checkpoint...
Eval Historical_best_acc = 0.8901 (at epoch 177)

Begin evaluation...
Is_training: False. [1,3],  running_mf1: 0.65947
acc: 0.80342 miou: 0.58053 mf1: 0.71499 iou_0: 0.77580 iou_1: 0.38526 F1_0: 0.87375 F1_1: 0.55622 precision_0: 0.93572 precision_1: 0.45119 recall_0: 0.81948 recall_1: 0.72500 

This is the result of my DSIFN dataset using two 2080 trained DSIFN datasets

**Comment by YangFanghan** [comment]
2536110410@qq.com,This is my email,Do you facilitate the e-mail to communicate some details?

**Comment by yuwanting828** [comment]
I'm actually not sure about the specifics, I'm also a beginner, but my image_size is 256 , as well as the dataset partitioning, I used the dataset partitioned in the link I sent you.

**Comment by shizizuoing** [comment]
> Can you share with me the training hyperparameters on the dsifn dataset, my training results are very different from the original text, and there is no best model after 60 epoch, this is my email, can you share the dsifn data collection, and your training methods and results。
Have you solved this problem? I have encountered similar problems。The test result of level_cd data set is much higher than that of the paper.
Can you share the super parameters with me? Thank you

**Comment by wulei1595** [comment]
> gpu_ids： [0， 3] project_name： CD_bit_b16_lr0.01 checkpoint_root： 检查点测试num_workers： 4 数据集： CDDataset data_name： DSIFN-CD-ori batch_size： 16 拆分： 火车split_val： 瓦尔 img_size： 512 n_class： 2 net_G： base_transformer_pos_s4_dd8_dedim8 损失： CE 优化器： 新加坡元 LR： 0.01 max_epochs： 200 lr_policy： 线性lr_decay_iters： 100 checkpoint_dir：检查点测试/CD_bit_b16_lr0.01 vis_dir：VIS/CD_bit_b16_lr0.01 加载最后一个检查点...评估Historical_best_acc = 0.8901（纪元 177 时）
> 
> 开始评估... Is_training：错。[1，3]， running_mf1： 0.65947 累积： 0.80342 MIOU： 0.58053 MF1： 0.71499 iou_0： 0.77580 iou_1： 0.38526 F1_0： 0.87375 F1_1： 0.55622 precision_0： 0.93572 precision_1： 0.45119 recall_0： 0.81948 recall_1： 0.72500
> 
> 这是我使用两个 2080 个经过训练的 DSIFN 数据集的 DSIFN 数据集的结果



> gpu_ids： [0， 3] project_name： CD_bit_b16_lr0.01 checkpoint_root： 检查点测试num_workers： 4 数据集： CDDataset data_name： DSIFN-CD-ori batch_size： 16 拆分： 火车split_val： 瓦尔 img_size： 512 n_class： 2 net_G： base_transformer_pos_s4_dd8_dedim8 损失： CE 优化器： 新加坡元 LR： 0.01 max_epochs： 200 lr_policy： 线性lr_decay_iters： 100 checkpoint_dir：检查点测试/CD_bit_b16_lr0.01 vis_dir：VIS/CD_bit_b16_lr0.01 加载最后一个检查点...评估Historical_best_acc = 0.8901（纪元 177 时）
> 
> 开始评估... Is_training：错。[1，3]， running_mf1： 0.65947 累积： 0.80342 MIOU： 0.58053 MF1： 0.71499 iou_0： 0.77580 iou_1： 0.38526 F1_0： 0.87375 F1_1： 0.55622 precision_0： 0.93572 precision_1： 0.45119 recall_0： 0.81948 recall_1： 0.72500
> 
> 这是我使用两个 2080 个经过训练的 DSIFN 数据集的 DSIFN 数据集的结果

请问您的数据集里的图片有无在训练前像论文里那样裁剪成256*256，还是直接用作者的代码缩放图片成256*256

**Comment by shizizuoing** [comment]
> 训练前像论文里那样裁剪成256_256，还是直接用作者的代码缩放图片成256_256

我是裁剪成256——256的，不是直接缩放。您的level-cd和原论文结果一样吗，我那个指标也很高

**Comment by wulei1595** [comment]
> > 训练前像论文里那样裁剪成256_256，还是直接用作者的代码缩放图片成256_256
> 
> 我是裁剪成256——256的，不是直接缩放。您的level-cd和原论文结果一样吗，我那个指标也很高

我没有裁剪，直接用的原数据集的1024*1024，那三个指标比论文里还低点。我现在才得到裁剪后的256-256数据集，现在准备试一下

**Comment by qianday** [comment]
> ###

请问您
使用裁剪后得到的结果是多少，我得到的超过94，比原论文过高，我不明白为什么会这样

**Comment by qianday** [comment]
> > > 训练之前像论文里那样裁剪成256_256，还是直接用作者的代码缩放图片成256_256
> > 
> > 
> > 我是理想成256——256的，不是直接缩放。你的level-cd和原论文结果一样吗，我那个指标也很好
> 
> 我没有裁剪，直接用原数据集的1024*1024，那三个指标比论文里还低点。我现在才得到裁剪后的256-256数据集，现在准备试一下

请问您裁剪之后得到的结果是多少，我得到的超过94，比原论文过高，我不明白为什么会这样
